src/Entity/LtiInstance.php

```diff
@@ -31,54 +31,45 @@ class Infrastructure implements EntityInterface
     private $label;
 
     /** @var string */
-    private $ltiDirectorLink;
+    private $ltiLink;
 
     /** @var string */
     private $ltiKey;
 
     /** @var string */
     private $ltiSecret;
 
-    public function getId(): int
+    public function __construct(int $id, string $label, string $ltiLink, string $ltiKey, string $ltiSecret)
     {
-        return $this->id;
-    }
-
-    public function getLabel(): string
-    {
-        return $this->label;
+        $this->id = $id;
+        $this->label = $label;
+        $this->ltiLink = $ltiLink;
+        $this->ltiKey = $ltiKey;
```

Comment:
Unfortunately no, since this is a doctrine entity.

Reply:
You can by Using [Embeddables](https://www.doctrine-project.org/projects/doctrine-orm/en/latest/tutorials/embeddables.html#separating-concerns-using-embeddables) :)

Reply:
Ah true, you are right, thx for the reminder :) Do you see any real value here of using embedded object? It would reduce the number of params by 1 that's true, but it would result in longer getters: Example: `$ltiInstance->getLtiCredentials()->getLtiSecret()`. Does it worth it?

Reply:
Both solutions are ok. I was only providing a suggestion to define better our domain and to be more objected-oriented adopting immutable concepts. I agree to you. we do not need this for now. :)